invoices/invoiceregistry.go

```diff
@@ -465,19 +598,95 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 			return nil, errNoUpdate
 		}
 
-		// If the invoice is already canceled, there is no further
-		// checking to do.
-		if inv.Terms.State == channeldb.ContractCanceled {
-			debugLog("invoice already canceled")
-			return nil, errNoUpdate
+		// Start building the accept descriptor.
+		acceptDesc := &channeldb.HtlcAcceptDesc{
+			Amt:          amtPaid,
+			Expiry:       expiry,
+			AcceptHeight: currentHeight,
 		}
 
-		// If an invoice amount is specified, check that enough
-		// is paid. Also check this for duplicate payments if
-		// the invoice is already settled or accepted.
-		if inv.Terms.Value > 0 && amtPaid < inv.Terms.Value {
-			debugLog("amount too low")
-			return nil, errNoUpdate
+		// If an invoice amount is specified and this is not an mpp,
+		// check that the exact amount is paid.
+		var canSettle bool
+		if mpp == nil {
+			// If the invoice is already canceled, there is no further
+			// checking to do.
+			if inv.Terms.State == channeldb.ContractCanceled {
+				debugLog("invoice already canceled")
+				return nil, errNoUpdate
+			}
+
+			// If an invoice amount is specified, check that enough
+			// is paid. Also check this for duplicate payments if
+			// the invoice is already settled or accepted.
+			if inv.Terms.Value > 0 && amtPaid < inv.Terms.Value {
+				debugLog("amount too low and no mpp")
+				return nil, errNoUpdate
+			}
+
+			// TODO(joostjager): Check invoice mpp required feature
```

Comment:
note to self, we won't need to assert the feature bit until we begin to make them mandatory in the invoice, so this can be deferred until we allow that

Reply:
Ok, updated comment.